core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -950,7 +958,7 @@ class ReplicaManager(val config: KafkaConfig,
   ): Unit = {
     // Skip verification if the request is not transactional or transaction verification is disabled.
     if (transactionalId == null ||
-      !config.transactionLogConfig.transactionPartitionVerificationEnable
+      (!config.transactionLogConfig.transactionPartitionVerificationEnable && transactionSupportedOperation != addPartition)
```

Comment:
Should we use `transactionSupportedOperation.isTV2` here?

Reply:
Yup -- there was a place in Calvin's PR I also need to update.